src/main/java/teammates/common/datatransfer/FeedbackQuestionAttributes.java

```diff
@@ -504,21 +507,30 @@ public void sanitizeForSaving() {
         this.courseId = Sanitizer.sanitizeTitle(courseId);
         this.creatorEmail = Sanitizer.sanitizeEmail(creatorEmail);
     }
+    
+    public boolean isJsonValid(String test) {
+        try {
+            new JSONObject(test);
+        } catch (JSONException ex) {
+            // edited, to include @Arthur's comment
+            // e.g. in case JSONArray is valid as well...
+            try {
+                new JSONArray(test);
+            } catch (JSONException exception) {
+                return false;
+            }
+        }
+        return true;
+    }
 
     /** 
      * This method converts the given Feedback*QuestionDetails object to JSON for storing
      * 
      * @param questionDetails
      */
     public void setQuestionDetails(FeedbackQuestionDetails questionDetails) {
-        // For Text questions, the questionText simply contains the question, not a JSON
-        // This is due to legacy data in the data store before there are multiple question types
-        if (questionDetails.getQuestionType() == FeedbackQuestionType.TEXT) {
-            questionMetaData = new Text(questionDetails.getQuestionText());
-        } else {
-            Gson gson = Utils.getTeammatesGson();
-            questionMetaData = new Text(gson.toJson(questionDetails, getFeedbackQuestionDetailsClass()));
-        }
+        Gson gson = Utils.getTeammatesGson();
+        questionMetaData = new Text(gson.toJson(questionDetails, getFeedbackQuestionDetailsClass()));
```

Comment:
previously for Text Questions we are simply saving question as Text, now saving that as Json string.